contracts/allocators/LUSDAllocator.sol

```diff
@@ -199,80 +217,143 @@ contract LUSDAllocator is Ownable {
         lqtyTokenAddress = _lqtyTokenAddress;
-
-        require(_stabilityPool != address(0), "stabilityPool address cannot be 0x0");
         lusdStabilityPool = IStabilityPool(_stabilityPool);
-
-        require(_lqtyStaking != address(0), "LQTY staking address cannot be 0x0");
         lqtyStaking = ILQTYStaking(_lqtyStaking);
-
         frontEndAddress = _frontEndAddress; // address can be 0
+        weth = IWETH(_wethAddress);
+        hopTokenAddress = _hopTokenAddress; // address can be 0
+        swapRouter = ISwapRouter(_uniswapV3Router);
+    }
+
+    /**
+        StabilityPool::withdrawFromSP() and LQTYStaking::stake() will send ETH here, so capture and emit the event
+     */
+    receive() external payable {
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    /* ======== CONFIGURE FUNCTIONS for Guardian only ======== */
+    function setEthToLUSDRatio(uint256 _ethToLUSDRatio) external onlyGuardian {
+        require(_ethToLUSDRatio <= 100 * FEE_PRECISION, "Value must be between 0 and 100 * 1e6");
+        ethToLUSDRatio = _ethToLUSDRatio;
+    }
+
+    function setPoolFee(uint256 _poolFee) external onlyGuardian {
+        require(_poolFee <= 100 * FEE_PRECISION, "Value must be between 0 and 100 * 1e6");
+        poolFee = _poolFee;
+    }
 
-        require(_wethAddress != address(0), "WETH token address cannot be 0x0");
-        weth = IERC20(_wethAddress);
+    function setHopTokenAddress(address _hopTokenAddress) external onlyGuardian {
+        hopTokenAddress = _hopTokenAddress;
+    }
+
+    /**
+     *  @notice setsFrontEndAddress for Stability pool rewards
+     *  @param _frontEndAddress address
+     */
+    function setFrontEndAddress(address _frontEndAddress) external onlyGuardian {
+        frontEndAddress = _frontEndAddress;
+    }
+
+    function updateTreasury() public onlyGuardian {
+        require(authority.vault() != address(0), "Zero address: Vault");
+        require(address(authority.vault()) != address(treasury), "No change");
+        treasury = ITreasury(authority.vault());
     }
 
     /* ======== OPEN FUNCTIONS ======== */
 
     /**
      *  @notice claims LQTY & ETH Rewards
+     * @param minETHLUSDRate minimum rate of when swapping ETH->LUSD.  e.g. 3500 means we swap at a rate of 1 ETH for a minimum 3500 LUSD
 
         1.  Harvest from LUSD StabilityPool to get ETH+LQTY rewards
         2.  Stake LQTY rewards from #1.  This txn will also give out any outstanding ETH+LUSD rewards from prior staking
-        3.  Deposit LUSD from #2 into StabilityPool
-        4.  Move ETH from #1 and #2 to treasury 
+        3.  If we have eth, convert to weth, then swap a percentage of it to LUSD.  If swap successul then send all remaining WETH to treasury
+        4.  Deposit LUSD from #2 and potentially #3 into StabilityPool
      */
-    function harvest() public returns (bool) {
+    function harvest(uint256 minETHLUSDRate) public onlyGuardian returns (bool) {
```

Comment:
Should annotate the decimals for the minETHLUSDRate in the comments.

Reply:
Done, see https://github.com/OlympusDAO/olympus-contracts/blob/v2-allocators-lusd4/contracts/allocators/LUSDAllocator.sol#L281